### ISSUE TITLE
Bug: Cannot lint both inside and outside code blocks

### ISSUE THREAD

**Issue by ehmicky** [eslint/markdown#297]
### Environment

ESLint version: 9.13.0
@eslint/markdown version: 6.2.1
Node version: 23.1.0
npm version: 10.9.0
Operating System: Ubuntu 24.04

### Which language are you using?

commonmark

## Configuration

```js
// eslint.config.js
import markdown from '@eslint/markdown'

export default [
  {
    plugins: {
      markdown,
    },
  },
  {
    files: ['**/*.md'],
    processor: 'markdown/markdown',
    language: 'markdown/commonmark',
    rules: {
      'markdown/no-duplicate-headings': 'error',
    }
  },
  {
    files: ['**/*.md/*.js'],
    rules: {
      'no-undef': 'error',
    },
  },
]
```

## Markdown file

````md
This is test.md

# Title

# Title

```js
test
```
````

## Actual behavior

```sh
$ ls
eslint.config.js  node_modules  package.json  package-lock.json  test.md
$ eslint

/home/ether/Desktop/try_eslint/test.md
  6:1  error  'test' is not defined  no-undef

✖ 1 problem (1 error, 0 warnings)
```

If I comment out the `processor: 'markdown/markdown'` line in my `eslint.config.js`, I get this instead:

```sh
$ eslint

/home/ether/Desktop/try_eslint/test.md
  3:1  error  Duplicate heading "Title" found  markdown/no-duplicate-headings

✖ 1 problem (1 error, 0 warnings)
```

### What did you expect to happen?

I expect to get both linting errors at the same time.

### What actually happened?

I either get linting applied to the contents outside of the code blocks, or inside of the code blocks, depending on whether `processor` is specified in my `eslint.config.js`.

### Link to Minimal Reproducible Example

https://github.com/ehmicky/eslint-markdown-bug

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by nzakas** [comment]
This is a limitation of ESLint itself, not with this plugin. We're working on it:
https://github.com/eslint/eslint/issues/14745

**Comment by ehmicky** [comment]
Hi @nzakas,

Thanks and sorry as this is a duplicate of https://github.com/eslint/markdown/issues/276.

It seems like the above issue might take a little time to resolve. Are there any known workarounds in the meantime?

**Comment by nzakas** [comment]
The only workaround is to run ESLint twice, one with each configuration. That's what this repo does.

**Comment by ehmicky** [comment]
Thanks for your answer @nzakas.
